### DISCUSSION TITLE
Creating Custom Execute Methods

### DISCUSSION THREAD

**Discussion by Gengaeldelse1** [fluentmigrator/fluentmigrator#1719] [Q&A]
I'm sorry if this has been asked before, but I would like to create some custom Execute methods for the project I'm working on.

What I'm trying to create are some methods that would allow my team members to save time in creating migrations by simplifying the Execute.Script() method.

What I would like for them to write is this:

Execute.Procedure("StoredProcedure");

Instead of:

Execute.Script(@"..\Scripts\Stored Procedures\StoredProcedure.sql");

Is this possible to do? If so, how?

**Comment by workgroupengineering** [comment]
See [AutoScriptMigration](https://github.com/fluentmigrator/fluentmigrator/blob/main/test%2FFluentMigrator.Tests%2FUnit%2FDefaultMigrationConventionsTests.cs#L662)

**Reply by Gengaeldelse1** [reply]
Unfortunately, this won't accomplish what I would like to do.  Due to my companies SQL code standards, I can't change the names of 2000 sql scripts to be able to use AutoScriptMigration.

I'm essentially wanting to write an extension class so I can pre fill the folder path for various objects.

Something like this:

    public class MigrationExtensions : ForwardOnlyMigration
    {
        public void ExecuteProcedure(string input)
        {
            Execute.Script($@"..\Scripts\Stored Procedures\{input}.sql");

        }
        
However, when I have a migration inherit from MigrationExtensions instead of ForwardOnlyMigrations, I get an error when I try to execute migrate.exe:

 Could not load file or assembly 'System.Runtime...

**Reply by workgroupengineering** [reply]
You can configure name convention implenting your IAutoNameConvention

[this](
https://github.com/fluentmigrator/fluentmigrator/blob/main/src%2FFluentMigrator.Runner.Core%2FConventions%2FDefaultAutoNameConvention.cs) is default implementation.

**Comment by PhenX** [comment]
@Gengaeldelse1 sorry for the very late reply, this will do the trick 

```csharp
using FluentMigrator.Builders.Execute;

public static class ExecuteExpressionRootExtensions
{
    public static void Procedure(this ExecuteExpressionRoot executeRoot, string procedureName)
    {
        executeRoot.Script($@"..\Scripts\Stored Procedures\{procedureName}.sql");
    }
}
```
